tools/CopyrightDateTool.scala

```diff
@@ -0,0 +1,171 @@
+    def fileLastUpdateYear: String
+    def gitLastUpdateYear: String
+    def fileCreationYear: String
+    def gitCreationYear: String
+
+    def isUpdateDateWrong(): Boolean = {
+      fileLastUpdateYear.toInt != gitLastUpdateYear.toInt
+    }
+
+    def isUpdateDateWrongStr(): Boolean = {
+      fileLastUpdateYear != gitLastUpdateYear
+    }
+  }
+
+  case class FileDates(pathToFile: String,
+                      fileLastUpdateYear: String,
+                      gitLastUpdateYear: String,
+                      fileCreationYear: String,
+                      gitCreationYear: String) extends DateComparison
+
+  case class GitInfo(message: String, date: String)
+
+  def readFilesList(): Array[String] = {
+    val git_files = Seq("/bin/sh", "-c", 
+      "git grep --name-only --fixed-strings" + 
+      " '|    (c) 20' -- '*.scala'|grep -v CopyrightDateTool").!!
+    val files_array = git_files.split("\n")
+    files_array
+  }
+
+  def readFile(pathToFile: String): String = {
+    val source = scala.io.Source.fromFile(pathToFile)("UTF-8")
+    val lines = try source.mkString finally source.close()
+    lines
+  }
+
+  def getGitCreationYear(pathToFile: String): String = {
+    val buf_dates = (s"git log   --format=%ci $pathToFile" !!)
+    val dates = buf_dates.split("\\n").distinct
+    val year_column = 0
+    val res = dates.last.split("-")(year_column)
+    res
+  }
+
+  def getGitLastUpdateYear(pathToFile: String): String = {
+    val buf_dates = (s"git log -3  --format=%cd $pathToFile" !!)
+    val buf_messages = (s"git log -3  $pathToFile" !!)
+    val first_empty_elem = 1
+    val messages = buf_messages.split("commit").distinct.drop(first_empty_elem)
+    val dates = buf_dates.split("\\n").distinct
+    val gits_info = (messages, dates).zipped.map{ case(a,b) => GitInfo(a,b)}
+    val res = gits_info.filter(!_.message.contains("[Sync copyright dates]"))
+    val year_column = 4
+    val first = 0
+    res(first).date.split(" ")(year_column)
+  }
+
+  def getYear(pathToFile: String, lower_bound: Int, upper_bound: Int): String = {
+    val file = readFile(pathToFile)
+    val third_line = 2
+    val start_index = file.split("\\n")(third_line).indexOf("(c) 20")
+    val year = file.split("\\n")(third_line)
+                   .substring(start_index + lower_bound,
+                              start_index + upper_bound)
+    year
+  }
+
+  def getFileCreationYear(pathToFile: String): String = {
+    val lb = "(c) ".length()
+    val ub = "(c) 20XX".length()
+    val fileCreationYear = getYear(pathToFile, lb, ub)
+    fileCreationYear
+  }
+
+  def getFileLastUpdateYear(pathToFile: String): String = {
+    val lb = "(c) 20XX-".length()
+    val ub = "(c) 20XX-XXXX".length()
+    val fileLastUpdateYear = getYear(pathToFile, lb, ub)
+    fileLastUpdateYear
+  }
+
+  def analyzeDates(): Array[FileDates] = {
+    val files_array = readFilesList()
+    var data = Array[FileDates]()
+    for(file_path <- files_array) {
```

Comment:
map over `files_array` instead of using a for loop

    val data = files_array map {
      case file_path =>
        val ...
        val ...
        val ...
        val ...
        FileDates(file_path, ...)
     }